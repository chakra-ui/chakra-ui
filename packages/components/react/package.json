{
  "name": "@chakra-ui/react",
  "version": "2.4.9",
  "description": "Responsive and accessible React UI components built with React and Emotion",
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@chakra-ui/accordion": "workspace:*",
    "@chakra-ui/alert": "workspace:*",
    "@chakra-ui/avatar": "workspace:*",
    "@chakra-ui/breadcrumb": "workspace:*",
    "@chakra-ui/button": "workspace:*",
    "@chakra-ui/card": "workspace:*",
    "@chakra-ui/checkbox": "workspace:*",
    "@chakra-ui/close-button": "workspace:*",
    "@chakra-ui/control-box": "workspace:*",
    "@chakra-ui/counter": "workspace:*",
    "@chakra-ui/css-reset": "workspace:*",
    "@chakra-ui/editable": "workspace:*",
    "@chakra-ui/form-control": "workspace:*",
    "@chakra-ui/hooks": "workspace:*",
    "@chakra-ui/icon": "workspace:*",
    "@chakra-ui/image": "workspace:*",
    "@chakra-ui/input": "workspace:*",
    "@chakra-ui/layout": "workspace:*",
    "@chakra-ui/live-region": "workspace:*",
    "@chakra-ui/media-query": "workspace:*",
    "@chakra-ui/menu": "workspace:*",
    "@chakra-ui/modal": "workspace:*",
    "@chakra-ui/number-input": "workspace:*",
    "@chakra-ui/pin-input": "workspace:*",
    "@chakra-ui/popover": "workspace:*",
    "@chakra-ui/popper": "workspace:*",
    "@chakra-ui/portal": "workspace:*",
    "@chakra-ui/progress": "workspace:*",
    "@chakra-ui/provider": "workspace:*",
    "@chakra-ui/radio": "workspace:*",
    "@chakra-ui/react-env": "workspace:*",
    "@chakra-ui/select": "workspace:*",
    "@chakra-ui/skeleton": "workspace:*",
    "@chakra-ui/slider": "workspace:*",
    "@chakra-ui/spinner": "workspace:*",
    "@chakra-ui/stat": "workspace:*",
    "@chakra-ui/styled-system": "workspace:*",
    "@chakra-ui/switch": "workspace:*",
    "@chakra-ui/system": "workspace:*",
    "@chakra-ui/table": "workspace:*",
    "@chakra-ui/tabs": "workspace:*",
    "@chakra-ui/tag": "workspace:*",
    "@chakra-ui/textarea": "workspace:*",
    "@chakra-ui/theme": "workspace:*",
    "@chakra-ui/theme-utils": "workspace:*",
    "@chakra-ui/toast": "workspace:*",
    "@chakra-ui/tooltip": "workspace:*",
    "@chakra-ui/transition": "workspace:*",
    "@chakra-ui/utils": "workspace:*",
    "@chakra-ui/visually-hidden": "workspace:*"
  },
  "peerDependencies": {
    "@emotion/react": "^11.0.0",
    "@emotion/styled": "^11.0.0",
    "framer-motion": ">=4.0.0",
    "react": ">=18",
    "react-dom": ">=18"
  },
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "license": "MIT",
  "homepage": "https://chakra-ui.com/",
  "repository": {
    "type": "git",
    "url": "https://github.com/chakra-ui/chakra-ui",
    "directory": "packages/components/react"
  },
  "keywords": [
    "react",
    "ui",
    "design-system",
    "react-components",
    "uikit",
    "accessible",
    "components",
    "emotion",
    "library",
    "design-system"
  ],
  "storybook": {
    "title": "Chakra UI",
    "url": "https://storybook.chakra-ui.com"
  },
  "devDependencies": {
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "framer-motion": "^6.2.9",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "clean-package": "2.1.1"
  },
  "scripts": {
    "build": "tsup src --dts",
    "dev": "pnpm build:fast -- --watch",
    "clean": "rimraf dist .turbo",
    "typecheck": "tsc --noEmit",
    "build:fast": "tsup src",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "clean-package": "../../../clean-package.config.json",
  "tsup": {
    "clean": true,
    "target": "es2019",
    "format": [
      "cjs",
      "esm"
    ]
  }
}
