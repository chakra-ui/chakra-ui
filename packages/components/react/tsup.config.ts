import fs from "node:fs"
import path from "node:path"
import { defineConfig } from "tsup"

function removeDirective(fileName: string) {
  const file = path.resolve(__dirname, "dist", fileName)
  return fs.promises.writeFile(
    file,
    fs.readFileSync(file, "utf-8").replace("'use client'\n", ""),
  )
}

export default defineConfig({
  format: ["cjs", "esm"],
  target: "es2019",
  sourcemap: true,
  banner: {
    js: "'use client'",
  },
  async onSuccess() {
    await Promise.all([
      removeDirective("index.mjs"),
      removeDirective("index.js"),
    ])
  },
})
