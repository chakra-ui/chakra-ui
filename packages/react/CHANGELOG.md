# Change Log

## 1.0.3

### Patch Changes

- [`0d0ed051`](https://github.com/chakra-ui/chakra-ui/commit/0d0ed0513ac1094833f1e0294f655af122682ff4)
  [#2707](https://github.com/chakra-ui/chakra-ui/pull/2707) Thanks
  [@TimKolberger](https://github.com/TimKolberger)! - Fix too narrow TypeScript
  type for theme override

- Updated dependencies
  [[`f3cc344e`](https://github.com/chakra-ui/chakra-ui/commit/f3cc344e1a7dd08dc7a312e6ce039d55a80cc526),
  [`72bbd0db`](https://github.com/chakra-ui/chakra-ui/commit/72bbd0dbb913ba38ee2b9191d12bf73713ae4398),
  [`653f3dd6`](https://github.com/chakra-ui/chakra-ui/commit/653f3dd6f30a17e366c069666acbfd9eddb11936),
  [`c73cce25`](https://github.com/chakra-ui/chakra-ui/commit/c73cce25bf986fd5b1261575ca395f731f827f53)]:
  - @chakra-ui/pin-input@1.1.0
  - @chakra-ui/theme@1.2.1
  - @chakra-ui/close-button@1.0.2
  - @chakra-ui/system@1.0.2
  - @chakra-ui/alert@1.0.2
  - @chakra-ui/toast@1.1.1
  - @chakra-ui/modal@1.2.1
  - @chakra-ui/popover@1.0.3
  - @chakra-ui/skeleton@1.0.2

## 1.0.2

### Patch Changes

- [`cfd3b325`](https://github.com/chakra-ui/chakra-ui/commit/cfd3b32564066076529811c5350aff6be565b7a3)
  [#2632](https://github.com/chakra-ui/chakra-ui/pull/2632) Thanks
  [@TimKolberger](https://github.com/TimKolberger)! - Improve typings for
  `extendTheme`. IntelliSense now can suggest property keys for colors and
  existing components.
- Updated dependencies
  [[`e1b095cd`](https://github.com/chakra-ui/chakra-ui/commit/e1b095cdb799bb3630d3cbb7891d8c7e929db0f4),
  [`6b527414`](https://github.com/chakra-ui/chakra-ui/commit/6b52741456f71ba4ec8cfe3e6307796d195adf5b),
  [`28bd21d7`](https://github.com/chakra-ui/chakra-ui/commit/28bd21d793911ba56bd146dd7aaff1008a70d147),
  [`2192be3d`](https://github.com/chakra-ui/chakra-ui/commit/2192be3db77504addffc89cbfef62fb9e884fa5d),
  [`a36ede75`](https://github.com/chakra-ui/chakra-ui/commit/a36ede7519b7193f90e4985636a06c5d483a8a62),
  [`ff10bcec`](https://github.com/chakra-ui/chakra-ui/commit/ff10bceca5774769766eef3a6812a22f387dd58d),
  [`ac0af38d`](https://github.com/chakra-ui/chakra-ui/commit/ac0af38d12dc5828503e178fb94a21f252e3a62b),
  [`c696345a`](https://github.com/chakra-ui/chakra-ui/commit/c696345a711338a23542a7b1911a33927a9ba5f1),
  [`9fdc61d8`](https://github.com/chakra-ui/chakra-ui/commit/9fdc61d8801f6d76783b5c9f068525d4dfc28b20),
  [`2416cf9a`](https://github.com/chakra-ui/chakra-ui/commit/2416cf9abe183a3a38adbccff794088d86a46341),
  [`58e26653`](https://github.com/chakra-ui/chakra-ui/commit/58e26653134caee1dd2caddb9014f7a90dc4eb7e)]:
  - @chakra-ui/number-input@1.0.2
  - @chakra-ui/layout@1.1.0
  - @chakra-ui/modal@1.2.0
  - @chakra-ui/theme@1.2.0
  - @chakra-ui/form-control@1.0.2
  - @chakra-ui/toast@1.1.0
  - @chakra-ui/transition@1.0.2
  - @chakra-ui/radio@1.0.2
  - @chakra-ui/input@1.0.2
  - @chakra-ui/select@1.0.2
  - @chakra-ui/textarea@1.0.2
  - @chakra-ui/accordion@1.0.2
  - @chakra-ui/menu@1.0.2
  - @chakra-ui/popover@1.0.2

## 1.0.1

### Patch Changes

- Updated dependencies
  [[`b3594802`](https://github.com/chakra-ui/chakra-ui/commit/b3594802714115c762013174badd5b838217be6f),
  [`843854ec`](https://github.com/chakra-ui/chakra-ui/commit/843854ec669367623b50a598402be343866d87a8),
  [`f0d03d9f`](https://github.com/chakra-ui/chakra-ui/commit/f0d03d9f6e474a1f1767643889734665c2910dc1),
  [`892ea2ca`](https://github.com/chakra-ui/chakra-ui/commit/892ea2ca1c02b4127f4f044df33de58cc7641f5c),
  [`5c482483`](https://github.com/chakra-ui/chakra-ui/commit/5c482483ce24fc798540c9792a15e06772eae213),
  [`204ff7e3`](https://github.com/chakra-ui/chakra-ui/commit/204ff7e39dd37003786194704b36069ef94d56a6),
  [`ba262ac7`](https://github.com/chakra-ui/chakra-ui/commit/ba262ac7b2e2d932bb227d4ff9181e83fbaa4149),
  [`a1ff404b`](https://github.com/chakra-ui/chakra-ui/commit/a1ff404b12a898ab97af024391a06c34da5bc69a)]:
  - @chakra-ui/layout@1.0.1
  - @chakra-ui/theme@1.1.0
  - @chakra-ui/progress@1.0.1
  - @chakra-ui/tabs@1.0.1
  - @chakra-ui/utils@1.0.1
  - @chakra-ui/hooks@1.0.1
  - @chakra-ui/modal@1.1.0
  - @chakra-ui/media-query@1.0.1
  - @chakra-ui/system@1.0.1
  - @chakra-ui/toast@1.0.1
  - @chakra-ui/accordion@1.0.1
  - @chakra-ui/alert@1.0.1
  - @chakra-ui/avatar@1.0.1
  - @chakra-ui/breadcrumb@1.0.1
  - @chakra-ui/button@1.0.1
  - @chakra-ui/checkbox@1.0.1
  - @chakra-ui/close-button@1.0.1
  - @chakra-ui/control-box@1.0.1
  - @chakra-ui/counter@1.0.1
  - @chakra-ui/editable@1.0.1
  - @chakra-ui/form-control@1.0.1
  - @chakra-ui/icon@1.0.1
  - @chakra-ui/image@1.0.1
  - @chakra-ui/input@1.0.1
  - @chakra-ui/live-region@1.0.1
  - @chakra-ui/menu@1.0.1
  - @chakra-ui/number-input@1.0.1
  - @chakra-ui/pin-input@1.0.1
  - @chakra-ui/popover@1.0.1
  - @chakra-ui/popper@1.0.1
  - @chakra-ui/portal@1.0.1
  - @chakra-ui/radio@1.0.1
  - @chakra-ui/select@1.0.1
  - @chakra-ui/skeleton@1.0.1
  - @chakra-ui/slider@1.0.1
  - @chakra-ui/spinner@1.0.1
  - @chakra-ui/stat@1.0.1
  - @chakra-ui/switch@1.0.1
  - @chakra-ui/tag@1.0.1
  - @chakra-ui/textarea@1.0.1
  - @chakra-ui/tooltip@1.0.1
  - @chakra-ui/transition@1.0.1
  - @chakra-ui/visually-hidden@1.0.1

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0 (2020-11-13)

**Note:** Version bump only for package @chakra-ui/react

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-rc.8 (2020-10-29)

### Bug Fixes

- **toast:** allow custom render in update
  ([eb8bff9](https://github.com/chakra-ui/chakra-ui/commit/eb8bff911e6ec9de0165ab1e8f5ca10d5e022459)),
  closes [#2362](https://github.com/chakra-ui/chakra-ui/issues/2362)

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-rc.7 (2020-10-25)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.6 (2020-10-25)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.5 (2020-09-27)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.4 (2020-09-25)

**Note:** Version bump only for package @chakra-ui/core

# 1.0.0-rc.3 (2020-08-30)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-rc.2 (2020-08-09)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-rc.1](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-rc.0...@chakra-ui/core@1.0.0-rc.1) (2020-08-06)

### Bug Fixes

- stack key issue and yarn2 deps issue
  ([d6cb6b8](https://github.com/chakra-ui/chakra-ui/commit/d6cb6b8fd964729efdf41b1e29c888a3c101316c))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-rc.0](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.7...@chakra-ui/core@1.0.0-rc.0) (2020-07-26)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.7](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.6...@chakra-ui/core@1.0.0-next.7) (2020-07-26)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.6](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.5...@chakra-ui/core@1.0.0-next.6) (2020-07-15)

**Note:** Version bump only for package @chakra-ui/core

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.5](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@1.0.0-next.4...@chakra-ui/core@1.0.0-next.5) (2020-07-15)

### Bug Fixes

- revert icon fallback to correct svg path
  ([afa4dbb](https://github.com/chakra-ui/chakra-ui/commit/afa4dbb49931391778f635784bddd3395d555df0))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.4](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.8.0...@chakra-ui/core@1.0.0-next.4) (2020-07-01)

### Bug Fixes

- [#891](https://github.com/chakra-ui/chakra-ui/issues/891)
  ([e107acc](https://github.com/chakra-ui/chakra-ui/commit/e107acc8487898a965b0d695c1da71f46fc56d5e))
- Adding PIN input to core export
  ([301483e](https://github.com/chakra-ui/chakra-ui/commit/301483e47996d266612f4f282611dfffca333bbc))
- ts issue with sx prop
  ([d3b1340](https://github.com/chakra-ui/chakra-ui/commit/d3b1340cb255937927b4d4c56ce218141570b951))
- **core:** add Skeleton export
  ([9255f24](https://github.com/chakra-ui/chakra-ui/commit/9255f24a3653f6f71773d4fdf7221c40aeddb873))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.3](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.8.0...@chakra-ui/core@1.0.0-next.3) (2020-06-28)

### Bug Fixes

- [#891](https://github.com/chakra-ui/chakra-ui/issues/891)
  ([e107acc](https://github.com/chakra-ui/chakra-ui/commit/e107acc8487898a965b0d695c1da71f46fc56d5e))
- ts issue with sx prop
  ([d3b1340](https://github.com/chakra-ui/chakra-ui/commit/d3b1340cb255937927b4d4c56ce218141570b951))
- **core:** add Skeleton export
  ([9255f24](https://github.com/chakra-ui/chakra-ui/commit/9255f24a3653f6f71773d4fdf7221c40aeddb873))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.2](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.8.0...@chakra-ui/core@1.0.0-next.2) (2020-06-21)

### Bug Fixes

- [#891](https://github.com/chakra-ui/chakra-ui/issues/891)
  ([e107acc](https://github.com/chakra-ui/chakra-ui/commit/e107acc8487898a965b0d695c1da71f46fc56d5e))
- **core:** add Skeleton export
  ([9255f24](https://github.com/chakra-ui/chakra-ui/commit/9255f24a3653f6f71773d4fdf7221c40aeddb873))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.4.0](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.4...@chakra-ui/core@0.4.0) (2019-10-14)

### Bug Fixes

- **box:** update type definitions
  ([b78723a](https://github.com/chakra-ui/chakra-ui/commit/b78723a))
- **css:** export css function
  ([c8a17c8](https://github.com/chakra-ui/chakra-ui/commit/c8a17c8))
- **cssreset:** remove comments from CSSReset preflight.js
  ([25edb5b](https://github.com/chakra-ui/chakra-ui/commit/25edb5b)), closes
  [#161](https://github.com/chakra-ui/chakra-ui/issues/161)
- **image:** add prop to opt out of the fallbackSrc logic
  ([cd1acc6](https://github.com/chakra-ui/chakra-ui/commit/cd1acc6))
- **image:** add support for native width and height
  ([e4b6165](https://github.com/chakra-ui/chakra-ui/commit/e4b6165))
- **image:** add support for native width and height
  ([298f067](https://github.com/chakra-ui/chakra-ui/commit/298f067))
- improve NumberInput API
  ([bca497e](https://github.com/chakra-ui/chakra-ui/commit/bca497e))
- **image:** improve image hook
  ([4fb5c04](https://github.com/chakra-ui/chakra-ui/commit/4fb5c04))
- **numberinput:** improve NumberInput API
  ([fe0441d](https://github.com/chakra-ui/chakra-ui/commit/fe0441d))
- **numberinput:** update API and docs for numberinput
  ([dc70274](https://github.com/chakra-ui/chakra-ui/commit/dc70274))
- **numberinput:** update story
  ([bda8efc](https://github.com/chakra-ui/chakra-ui/commit/bda8efc))
- **types:** add types for default theme. closes issue
  [#160](https://github.com/chakra-ui/chakra-ui/issues/160)
  ([dfef504](https://github.com/chakra-ui/chakra-ui/commit/dfef504))
- **types:** add types for icons to improve DX
  ([5629bc7](https://github.com/chakra-ui/chakra-ui/commit/5629bc7))
- **types:** update types to allow for migration to TS
  ([73bbe73](https://github.com/chakra-ui/chakra-ui/commit/73bbe73))

### Features

- add SimpleGrid component
  ([ff5760c](https://github.com/chakra-ui/chakra-ui/commit/ff5760c))
- add SimpleGrid component and improve Stack
  ([16cc9ef](https://github.com/chakra-ui/chakra-ui/commit/16cc9ef))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.4](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.3...@chakra-ui/core@0.3.4) (2019-10-09)

### Bug Fixes

- **build:** remove babel-runtime dep to fix modal issue
  ([f4ae47f](https://github.com/chakra-ui/chakra-ui/commit/f4ae47f))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.3](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.2...@chakra-ui/core@0.3.3) (2019-10-09)

### Bug Fixes

- **box:** add overflow types from styled-system
  ([805cb61](https://github.com/chakra-ui/chakra-ui/commit/805cb61))
- **box:** update type definition for borderWidth props
  ([f4310ec](https://github.com/chakra-ui/chakra-ui/commit/f4310ec))
- **button:** fix the types for leftIcon and rightIcon
  ([9f3ba3a](https://github.com/chakra-ui/chakra-ui/commit/9f3ba3a))
- **cssreset:** update the type definition
  ([d8f2701](https://github.com/chakra-ui/chakra-ui/commit/d8f2701))
- **iconbutton:** fix the types for the iconbutton
  ([b9462d4](https://github.com/chakra-ui/chakra-ui/commit/b9462d4))
- **modal:** add support for preserving scrollbar gap
  ([bd67ea0](https://github.com/chakra-ui/chakra-ui/commit/bd67ea0))
- **types:** export named types in all components
  ([7378b77](https://github.com/chakra-ui/chakra-ui/commit/7378b77))
- resolve yarn v2 pnp issues
  ([0e0a04f](https://github.com/chakra-ui/chakra-ui/commit/0e0a04f))
- **types:** include named exports from theme in main type definitions
  ([68e06d8](https://github.com/chakra-ui/chakra-ui/commit/68e06d8))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.2](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.1...@chakra-ui/core@0.3.2) (2019-09-26)

### Bug Fixes

- **alertdialog:** update type definitions
  ([da3c706](https://github.com/chakra-ui/chakra-ui/commit/da3c706))
- **drawer:** update type definitions
  ([290e82c](https://github.com/chakra-ui/chakra-ui/commit/290e82c))
- **input:** resolve the error and focus border color props
  ([45b0d97](https://github.com/chakra-ui/chakra-ui/commit/45b0d97))
- **popover:** update type definitions
  ([3a221a3](https://github.com/chakra-ui/chakra-ui/commit/3a221a3))
- **tooltip:** fix the shouldwrapchildren issues. fixes issue
  [#121](https://github.com/chakra-ui/chakra-ui/issues/121)
  ([5d90b77](https://github.com/chakra-ui/chakra-ui/commit/5d90b77))

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.1](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.3.0...@chakra-ui/core@0.3.1) (2019-09-25)

### Bug Fixes

- aliases and forwarding
  ([cec87c6](https://github.com/chakra-ui/chakra-ui/commit/cec87c6))
- compose extraConfig
  ([07c9c2a](https://github.com/chakra-ui/chakra-ui/commit/07c9c2a))
- resolve npm pack issue
  ([8b0676e](https://github.com/chakra-ui/chakra-ui/commit/8b0676e))
- resolve npm pack issue
  ([44b299f](https://github.com/chakra-ui/chakra-ui/commit/44b299f))

# 0.3.0-beta (2019-09-25)

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/core@0.2.8...@chakra-ui/core@0.3.0) (2019-09-24)

### Features

- Add typings for Slider.onChange
  ([2f15c6a](https://github.com/chakra-ui/chakra-ui/commit/2f15c6a)), closes
  [chakra-ui/chakra-ui#88](https://github.com/chakra-ui/chakra-ui/issues/88)
- **modal, drawer, alertdialog:** remove console errors and improve api
  ([c5ce4b4](https://github.com/chakra-ui/chakra-ui/commit/c5ce4b4))
