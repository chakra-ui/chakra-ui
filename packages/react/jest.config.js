const baseConfig = require("../../jest.config")

module.exports = {
  ...baseConfig,
}
