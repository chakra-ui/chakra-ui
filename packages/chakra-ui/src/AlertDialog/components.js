/** @jsx jsx */
import { jsx } from "@emotion/core";
import styled from "@emotion/styled";
import {
  AlertDialogContent as ReachAlertDialogContent,
  AlertDialogDescription,
  AlertDialogLabel,
  AlertDialogOverlay as ReachAlertDialogOverlay,
} from "@reach/alert-dialog";
import { forwardRef } from "react";
import Box, { systemProps } from "../Box";
import extraConfig from "../Box/config";
import { useColorMode } from "../ColorModeProvider";
import { modalContentStyle } from "../Modal";

const AlertDialogHeader = props => (
  <Box py={4} px={6} as={AlertDialogLabel} {...props} />
);

const AlertDialogBody = props => (
  <Box px={6} py={2} as={AlertDialogDescription} {...props} />
);

const AlertDialogFooter = props => <Box ml="auto" p={6} {...props} />;

const StyledOverlay = styled(ReachAlertDialogOverlay)(systemProps, extraConfig);

const AlertDialogOverlay = ({
  zIndex = 5,
  bg = "rgba(16,22,26,.7)",
  ...props
}) => (
  <StyledOverlay
    data-testid="alert-dialog-overlay"
    position="fixed"
    zIndex={zIndex}
    bottom="0"
    top="0"
    left="0"
    right="0"
    overflowY="auto"
    bg={bg}
    {...props}
  />
);

const AlertDialogContent = forwardRef((props, ref) => {
  const { colorMode } = useColorMode();
  const styleProps = modalContentStyle({ colorMode });

  return (
    <Box
      as={ReachAlertDialogContent}
      width="100%"
      position="relative"
      display="flex"
      flexDirection="column"
      ref={ref}
      {...styleProps}
      {...props}
    />
  );
});

export {
  AlertDialogHeader,
  AlertDialogBody,
  AlertDialogFooter,
  AlertDialogOverlay,
  AlertDialogContent,
};
