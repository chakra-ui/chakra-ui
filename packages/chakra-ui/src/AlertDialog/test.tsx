import * as React from "react";
import {
  render,
  fireEvent,
  waitForElement,
  waitForElementToBeRemoved,
  prettyDOM,
} from "../utils/testing";
import AlertDialog from "./index";
import {
  AlertDialogHeader,
  AlertDialogBody,
  AlertDialogFooter,
} from "./components";

const MockDialog: React.FunctionComponent = () => {
  const [isOpen, setOpen] = React.useState(false);
  const cancelRef = React.useRef(null);
  const onClose = jest.fn(() => setOpen(false));
  const onConfirm = jest.fn(onClose);

  return (
    <>
      <button onClick={() => setOpen(true)}>open</button>
      <AlertDialog
        isOpen={isOpen}
        leastDestructiveRef={cancelRef}
        onClose={onClose}
      >
        <AlertDialogHeader>Dialog Header</AlertDialogHeader>

        <AlertDialogBody>Dialog content</AlertDialogBody>

        <AlertDialogFooter>
          <button ref={cancelRef} onClick={onClose}>
            Cancel
          </button>
          <button onClick={onConfirm}>Confirm</button>
        </AlertDialogFooter>
      </AlertDialog>
    </>
  );
};

it("should open and close", async () => {
  const { queryByText, getByText } = render(<MockDialog />);
  // Should start closed
  expect(queryByText("Dialog Header")).toBeNull();
  // Open the dialog
  fireEvent.click(getByText("open"));
  // Wait for it to appear
  await waitForElement(() => getByText("Dialog Header"));
  // Content should be there
  getByText("Dialog content");
  // Focus shoudld be on the cancel button
  expect(document.activeElement).toEqual(getByText("Cancel"));
  // Click on the cancel button
  fireEvent.click(getByText("Cancel"));
  // Wait for the dialog to disappear
  await waitForElementToBeRemoved(() => queryByText("Dialog Header"));
});

it("should match snapshot when open", async () => {
  const { getByText, getByTestId } = render(<MockDialog />);
  fireEvent.click(getByText("open"));
  await waitForElement(() => getByText("Dialog Header"));
  const dialog = getByTestId("alert-dialog-overlay");
  expect(dialog).toMatchSnapshot();
});

it("should close when you press escape", async () => {
  const { findByText, getByText } = render(<MockDialog />);
  fireEvent.click(getByText("open"));
  await findByText("Dialog Header");
  fireEvent.keyDown(getByText("Dialog content"), {
    key: "Escape",
    code: 27,
  });
  await waitForElementToBeRemoved(() => getByText("Dialog Header"));
});

it("should close when you click on the overlay", async () => {
  const { getByTestId, findByText, getByText } = render(<MockDialog />);
  fireEvent.click(getByText("open"));
  await findByText("Dialog Header");
  const overlay = getByTestId("alert-dialog-overlay");
  fireEvent.click(overlay);
  await waitForElementToBeRemoved(() => getByText("Dialog Header"));
});

it("should not close when you click inside the alert dialog", async () => {
  const { getByText } = render(<MockDialog />);
  fireEvent.click(getByText("open"));
  await waitForElement(() => getByText("Dialog Header"));
  getByText("Dialog Header");
});

it("should have correct role and aria attributes", async () => {
  const { getByText, getByRole } = render(<MockDialog />);
  fireEvent.click(getByText("open"));
  await waitForElement(() => getByText("Dialog Header"));
  const dialog = getByRole("alertdialog");
  expect(dialog).toHaveAttribute("aria-modal", "true");
});
