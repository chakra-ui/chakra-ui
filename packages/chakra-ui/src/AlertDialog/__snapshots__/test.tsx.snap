// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when open 1`] = `
.emotion-0 {
  padding-top: 1rem;
  padding-bottom: 1rem;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.emotion-1 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.emotion-2 {
  margin-left: auto;
  padding: 1.5rem;
}

.emotion-4 {
  position: fixed;
  z-index: 5;
  bottom: 0;
  top: 0;
  left: 0;
  right: 0;
  overflow-y: auto;
  background-color: rgba(16,22,26,.7);
  overflow-y: auto;
  opacity: 0.2;
}

.emotion-3 {
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  color: inherit;
  box-shadow: 0 7px 14px 0 rgba(0,0,0,0.1),0 3px 6px 0 rgba(0,0,0,.07);
  max-width: 28rem;
  top: 10rem;
  margin-left: auto;
  margin-right: auto;
  border-radius: 0.25rem;
  -webkit-transform: translate3d(0px,8px,0px);
  -ms-transform: translate3d(0px,8px,0px);
  transform: translate3d(0px,8px,0px);
}

<div
  bg="rgba(16,22,26,.7)"
  bottom="0"
  class="emotion-4"
  data-reach-alert-dialog-overlay="true"
  data-reach-dialog-overlay="true"
  data-testid="alert-dialog-overlay"
  left="0"
  overflowy="auto"
  position="fixed"
  right="0"
  top="0"
  zindex="5"
>
  <div
    aria-labelledby="alert-dialog-null"
    aria-modal="true"
    class="emotion-3"
    data-reach-alert-dialong-content="true"
    data-reach-dialog-content="true"
    role="alertdialog"
    tabindex="-1"
    transform="translate3d(0px, 8px, 0px)"
  >
    <div
      class="emotion-0"
      data-reach-alert-dialog-label="true"
      id="alert-dialog-null"
    >
      Dialog Header
    </div>
    <div
      class="emotion-1"
      data-reach-alert-dialog-description="true"
      id="alert-dialog-null"
    >
      Dialog content
    </div>
    <div
      class="emotion-2"
    >
      <button>
        Cancel
      </button>
      <button>
        Confirm
      </button>
    </div>
  </div>
</div>
`;
